modules/popup/react/spec/Popup.spec.tsx

```diff
@@ -1,25 +1,84 @@
 import * as React from 'react';
+import {render, fireEvent} from '@testing-library/react';
 import Popup from '../lib/Popup';
-import {mount} from 'enzyme';
 
 describe('Popup', () => {
   const cb = jest.fn();
   afterEach(() => {
     cb.mockReset();
   });
 
-  test('should call a callback function', () => {
-    const component = mount(<Popup handleClose={cb}>Hello World</Popup>);
-    const popup = component.find('button');
-    popup.simulate('click');
-    expect(cb.mock.calls.length).toBe(1);
-    component.unmount();
+  describe('when rendered with a heading', () => {
+    it('should render popup with heading text', () => {
+      const headingText = 'Delete item';
+      const {getByRole} = render(
+        <Popup heading={headingText}>
+          <div>Are you sure you'd like to delete the item titled 'My Item'?</div>
+
+          <button onClick={cb}>Delete</button>
+        </Popup>
+      );
+      expect(getByRole('dialog').querySelector('h3')).toContainHTML(headingText);
+    });
+  });
+
+  describe('when rendered with a handleClose', () => {
+    it('should render popup with a close button', () => {
+      const closeButtonAriaLabel = 'close';
+      const {getAllByRole} = render(
+        <Popup closeLabel={closeButtonAriaLabel} handleClose={cb}>
+          <div>Are you sure you'd like to delete the item titled 'My Item'?</div>
+
+          <button onClick={cb}>Delete</button>
+        </Popup>
+      );
+
+      expect(getAllByRole('button')[0]).toHaveAttribute('aria-label', closeButtonAriaLabel);
```

Comment:
It looks like we added `data-close` as an attribute to help with selecting. Would that be a safer and more helpful selector here?

Reply:
i think that makes sense, I can make that change!